### PR TITLE
move rest-utils to ak configdefs

### PR DESCRIPTION
**Pull request by norwood**

### CODE REVIEW

**Comment by norwood** [comment]
tagged ya'll since thats what github told me to do, and it will affect all of us.

File: core/src/main/java/io/confluent/rest/Application.java [hunk 1]
```diff
@@ -191,7 +191,16 @@ public Server createServer() throws ServletException {
...
     ResourceConfig resourceConfig = new ResourceConfig();
 
-    Map<String, String> configuredTags = getConfiguration().getMap(RestConfig.METRICS_TAGS_CONFIG);
+    Map<String, String> configuredTags = new HashMap<>();
+    // This is copied from the old MAP implementation from cp ConfigDef.Type.MAP
+    List<String> metricsTags = getConfiguration().getList(RestConfig.METRICS_TAGS_CONFIG);
+    for (String entry : getConfiguration().getList(RestConfig.METRICS_TAGS_CONFIG)) {
+      String[] keyValue = entry.split("\\s*:\\s*", -1);
+      if (keyValue.length != 2) {
+        throw new ConfigException("Map entry should be of form <key>:<value");
```
Comment (woshifudayun) [inline core/src/main/java/io/confluent/rest/Application.java]:
```suggestion
        throw new ConfigException("Map entry should be of form <key>:<value>");
```

**Comment by norwood** [comment]
@ewencp if people are 👍 here i will make prs for all downstream repos and then we can merge all at once 😰

File: core/src/main/java/io/confluent/rest/Application.java [hunk 2]
```diff
@@ -191,7 +191,16 @@ public Server createServer() throws ServletException {
     // The configuration for the JAX-RS REST service
     ResourceConfig resourceConfig = new ResourceConfig();
 
-    Map<String, String> configuredTags = getConfiguration().getMap(RestConfig.METRICS_TAGS_CONFIG);
+    Map<String, String> configuredTags = new HashMap<>();
+    // This is copied from the old MAP implementation from cp ConfigDef.Type.MAP
+    List<String> metricsTags = getConfiguration().getList(RestConfig.METRICS_TAGS_CONFIG);
```
Comment (big-andy-coates) [inline core/src/main/java/io/confluent/rest/Application.java]: nit: not used.

File: core/src/main/java/io/confluent/rest/Application.java [hunk 3]
```diff
@@ -191,7 +191,16 @@ public Server createServer() throws ServletException {
...
+    for (String entry : getConfiguration().getList(RestConfig.METRICS_TAGS_CONFIG)) {
```
Comment (big-andy-coates) [inline core/src/main/java/io/confluent/rest/Application.java]: Probably worth putting this into a utility method.

Comment (norwood) [reply core/src/main/java/io/confluent/rest/Application.java]: also added a test

**Comment by big-andy-coates** [comment]
@norwood any thoughts about just adding a MAP type on Apache ConfigDef?

**Comment by norwood** [comment]
yeah, i'd be down to move it. but i'd rather not gate this pr on moving something to AK 👀

**Pull request merged**